pkg/controller/storagecluster/external_resources.go

```diff
@@ -171,6 +172,28 @@ func (r *ReconcileStorageCluster) createExternalStorageClusterResources(scs []*s
 		}
 		objectKey := types.NamespacedName{Name: d.Name, Namespace: instance.Namespace}
 		switch d.Kind {
+		case "CephCluster":
+			monitoringIP, ok := d.Data["MonitoringEndpoint"]
+			if !ok {
+				err := fmt.Errorf(
+					"Monitoring Endpoint not present in the external cluster secret %s",
+					externalClusterDetailsSecret)
+				reqLogger.Error(err, err.Error())
+				return err
+			}
+			monitoringPort, ok := d.Data["MonitoringPort"]
+			if !ok {
+				err := fmt.Errorf(
+					"Monitoring Port not present in the external cluster secret %s",
+					externalClusterDetailsSecret)
+				reqLogger.Error(err, err.Error())
+				return err
+			}
+			err := validateMonitoringEndpoint(monitoringIP, monitoringPort, reqLogger)
+			if err != nil {
+				reqLogger.Error(err, "Monitoring validation failed.")
+				return err
+			}
```

Comment:
This whole thing looks like a duplicate of `getExternalMonitoringInfo()`. Can we move everything there (including validation) there? I don't see anything being created with IP or Ports here.

Reply:
I will update it, The port is required to test the endpoint reachability, though we don't pass it in the Monitoring Spec. Rook already has the default port set. It is a safety net for the case if user enters ports different from(very rare) rook and tell them they need to run it on the default RHCS port.